components/VM_Arm/src/main.c

```diff
@@ -620,6 +620,29 @@ static void irq_handler(void *data, ps_irq_acknowledge_fn_t acknowledge_fn, void
     }
 }
 
+/* Default RAM initialization. Modules can overwrite this weak function with a
+ * custom initialization. e.g. to use memory shared with other components or
+ * mapped on demand.
+ */
+WEAK int init_ram_module(vm_t *vm)
+{
+    if (0 == ram_size) {
+        ZF_LOGW("VM%d: no RAM defined", vm->vm_id);
+        return 0;
+    }
+
+    ZF_LOGI("VM%d: %p - %p (RAM)", vm->vm_id, (void *)ram_base,
```

Comment:
1. Perhaps remove `VM%d: ` prefix as discussed above.
2. Consider using `"0x%"PRIxPTR` instead of `%p`. Then no  need to cast to `(void *)`.
3. How about reporting last valid address instead of first unvalid address?
4. Consistency: error message would complain about registration, but this is just two hexadecimal values?